scripts/ci/kubernetes/docker/build.sh

```diff
@@ -34,7 +34,7 @@ fi
 echo "Airflow directory $AIRFLOW_ROOT"
 echo "Airflow Docker directory $DIRNAME"
 
-if [[ ${PYTHON_VERSION} == '3' ]]; then
+if [[ ${PYTHON_VERSION} == 3* ]]; then
```

Comment:
Does this mean we actually end up running the kube tests against python 3.6 twice (once for 3.5 and 3.6 in the matrix?)

Reply:
We can actually choose what set of python/kube/env combinations we want to use and we will be able to choose what is the best matrix. Currently I added both 3.5 and 3.6 - mostly because I know 3.6 vs. 3.5 might differ but also to test how it works with more than 2/3 separation. But in the final version we might deliberately choose what to run and we might decide not to run either 3.6 or 3.5 or whichever combo we want. I found that combining full matrix + excludes is a bit easier and nicer than what we had before (see .travis.yml).

Reply:
One interesting thing on 3.5 vs 3.6 that I noticed - It's probably worth testing against 3.5.2 explicitly as that is the version of Python3 on Ubuntu 16.04 (and maybe 18.04 too?) - one of our downstream deps had an issue with 3.5.2, so testing that explicitly makes sense as it's a common deployment target.

Reply:
I will keep it in mind and change the scripts to allow x.y.z as well. Currently I rely on having x.y version being 3 chars but i will make it more general. It should be as easy as defining the right version to build in DockerHub.

However 3.5.2 might be slightly problematic on its own. 

There are no official python-slim images for 3.5.2 (only 3.5.6 from 3.5 series is available: https://hub.docker.com/_/python/ ) so we might have to either prepare our own image or find reliable (and safe) source for that. There are Docker files we can use from official python repo so that is not a big problem, it's just a matter of putting that to a registry somewhere:  https://github.com/docker-library/python/blob/af2cf72d9c6c304d041c88db3e79242fd8c9ce60/3.5/stretch/slim/Dockerfile 

I just checked default versions:
Xenial: 16.0.4 has 3.5.2 and 2.7.12
Bionic: 18.04 has 3.6.7 and 2.7.15

Reply:
I will keep it open. I think we will have to decide what i the matrix we are going to use eventually (I left TODO in https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-10+Multi-layered+and+multi-stage+official+Airflow+image)